src/ArrangerConduit.sol

```diff
@@ -0,0 +1,251 @@
+
+    function drawFunds(address asset, uint256 amount) external override isFundManager {
+        outstandingPrincipal[asset] += amount;
+
+        require(
+            amount <= ERC20Like(asset).balanceOf(address(this)) - totalWithdrawable[asset],
+            "Conduit/insufficient-available-cash"
+        );
+
+        require(ERC20Like(asset).transfer(fundManager, amount), "Conduit/transfer-failed");
+    }
+
+    function returnFunds(address asset, uint256 returnAmount) external override isFundManager {
+        outstandingPrincipal[asset] -= returnAmount;
+
+        uint256 fundsRemaining = returnAmount;
+
+        uint256 newStartingFundRequestId = startingFundRequestId[asset];
+
+        // For all of the assets fund requests, fill as much as possible
+        // Maintain the order of the fund requests array and update after all fills are complete.
+        for (uint256 i = newStartingFundRequestId; i < fundRequests[asset].length; i++) {
+            FundRequest storage fundRequest = fundRequests[asset][i];
+
+            if (
+                fundRequest.status == StatusEnum.CANCELLED ||
+                fundRequest.status == StatusEnum.COMPLETED
+            ) {
+                newStartingFundRequestId++;
+                continue;
+            }
+
+            uint256 fillAmount = fundRequest.amountRequested - fundRequest.amountFilled;
+
+            if (fillAmount > fundsRemaining) {
+                fillAmount = fundsRemaining;
+                fundRequest.status = StatusEnum.PARTIAL;
+            } else {
+                fundRequest.status = StatusEnum.COMPLETED;
+                newStartingFundRequestId++;
+            }
+
+            fundsRemaining           -= fillAmount;
+            fundRequest.amountFilled += fillAmount;
+            totalWithdrawable[asset] += fillAmount;
+
+            maxWithdraw[fundRequest.ilk][asset] += fillAmount;
+
+            if (fundsRemaining == 0) break;
+        }
+
+        startingFundRequestId[asset] = newStartingFundRequestId;
+
+        require(
+            ERC20Like(asset).transferFrom(fundManager, address(this), returnAmount),
+            "Conduit/transfer-failed"
+        );
+    }
+
+    function payInterest(address asset, uint256 amount) external override isFundManager {
+        totalInterestEarned[asset] += amount;
+
+        require(
+            ERC20Like(asset).transferFrom(fundManager, admin, amount),
+            "Conduit/transfer-failed"
+        );
+    }
+
+    /**********************************************************************************************/
+    /*** View Functions                                                                         ***/
+    /**********************************************************************************************/
+
+    function maxDeposit(bytes32 ilk, address asset)
+        external override pure returns (uint256 maxDeposit_)
+    {
+        ilk; asset;  // Silence warnings
```

Comment:
Can just omit the names in the arguments too. `function maxDeposit(bytes32, address)`

Reply:
This is still an issue.

Reply:
Updated